### PR TITLE
Add an interface to accept parsed html data.

### PR DESCRIPTION
**Pull request by dbalan**
Use parsed html data (`html.Node`) to generate text.

### CODE REVIEW

**Comment by dbalan** [comment]
@jaytaylor Would you be interested in merging this?

**Comment by jaytaylor** [comment]
@dbalan Thanks for following up.  I'll review it now.

File: html2text.go [hunk 1]
```diff
@@ -273,22 +273,26 @@ func getAttrVal(node *html.Node, attrName string) string {
 	return ""
 }
 
-func FromReader(reader io.Reader) (string, error) {
-	doc, err := html.Parse(reader)
-	if err != nil {
-		return "", err
-	}
-
+func FromHtmlRoot(doc *html.Node) (string, error) {
```
Comment (jaytaylor) [inline html2text.go]: Can you change `FromHtmlRoot` to `FromHTMLNode` to ensure optimal clarity?

**Comment by jaytaylor** [comment]
@dbalan I'd love to merge this, as soon as the function name is changed and there is a unit-test to cover this beautiful new functionality!

Cheers

File: html2text.go (hunk 1 above)
Comment (dbalan) [inline html2text.go]: agree :+1:

**Comment by dbalan** [comment]
@jaytaylor both the APIs already being tested `FromString` and `FromReader` calls `FromHtmlNode`. I'm not sure whats left to test without duplicating the current ones.

**Comment by jaytaylor** [comment]
@dbalan That's a good point, thank you!

**Pull request merged**